sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/DoubleExponentialHistogramBuckets.java

```diff
@@ -234,19 +257,44 @@ public boolean equals(@Nullable Object obj) {
     DoubleExponentialHistogramBuckets other = (DoubleExponentialHistogramBuckets) obj;
     // Don't need to compare getTotalCount() because equivalent bucket counts
     // imply equivalent overall count.
-    return getBucketCounts().equals(other.getBucketCounts())
-        && this.getOffset() == other.getOffset()
-        && this.scale == other.scale;
+    return this.scale == other.scale && sameBucketCounts(other);
```

Comment:
Why have you omitted the offset from `hashCode()` and `equals()`? I would think offset needs to be the same for two bucket sets to be strictly equal.

Reply:
exactly the case.  Additionally the other bucket needs to have ONLY zeros after the first bucket set ends.

We could fix this in the `diff` method with a "normalize" type method that ensures the first index is populated.   I'm fine going that route if you think it's cleaner.

Reply:
Definitely worth a comment in here, either way.

Reply:
Yeah agreed, could use a comment.

Reply:
Great point. I had documented the other method, but in-context, it wasn't very good.  Added here.